### ISSUE TITLE
BitVector encoder doesn't preserve length

### ISSUE THREAD

**Issue by durban** [circe/circe#551]
The encoder for `scodec.bits.BitVector` doesn't preserve the length of the bit vector (if it isn't a multiple of 8), but right pads it with zero bits (to be a multiple of 8). Thus, encoding and decoding is not (always) an identity.

```scala
scala> import scodec.bits._
import scodec.bits._

scala> import io.circe.syntax._
import io.circe.syntax._

scala> import io.circe.scodec._
import io.circe.scodec._

scala> val b = bin"010110"
b: scodec.bits.BitVector = BitVector(6 bits, 0x58)

scala> b.asJson.as[BitVector]
res0: io.circe.Decoder.Result[scodec.bits.BitVector] = Right(BitVector(8 bits, 0x58))
```

**Comment by travisbrown** [comment]
Eek! Thanks for catching this. I probably won't get around to fixing it until Sunday, if someone else wants to take a look before then. Once it's done I'll publish an 0.7.1 release.

**Comment by note** [comment]
Made a simple fix on my branch, still need to do some research as it's not perfect. Will get back to this tomorrow

**Comment by travisbrown** [comment]
@note Thansk! I guess it's obvious that this totally slipped off my radar in January…

**Comment by durban** [comment]
Fixed in 0.8.0.
